std/math.d

```diff
@@ -4103,9 +4103,55 @@ long lrint(real x) @trusted pure nothrow @nogc
 
             return sign ? -result : result;
         }
+        else static if (F.realFormat == RealFormat.ieeeQuadruple)
+        {
+            long result;
+
+            // Rounding limit when casting from real(128-bit) to ulong.
+            enum real OF = 0x1P+112;
+
+            ulong* vl = cast(ulong*)(&x);
+
+            // Find the exponent and sign
+            ulong msb = vl[MANTISSA_MSB];
+            ulong lsb = vl[MANTISSA_LSB];
+            int exp = cast(int)((msb >> 48) & 0x7fff) - 0x3fff;
+            int sign = cast(int)(msb >> 63);
+            msb &= 0xffff_ffffffff;
+            msb |= 0x10000_00000000;
+
+            if (exp < 63)
+            {
+                // Adjust x and check result.
+                real j = sign ? -OF : OF;
+                x = (j + x) - j;
```

Comment:
Can we get a floating-point expert to explain this for us? @ibuclaw or @9il ?

I'd prefer that we not just "cargo cult" this without knowing exactly what it does/if we really need it for `ieeeQuadruple`.